assets/app/scripts/services/data.js

```diff
@@ -559,7 +559,31 @@ angular.module('openshiftConsole')
     }
   };
 
+  function itemKindForList(data) {
+    var pos, kind = data.kind;
+    if (kind) {
+      pos = kind.lastIndexOf("List");
+      if (pos === kind.length - 4)
+        return kind.substring(0, pos);
+    }
+    return null;
+  }
+
   DataService.prototype._listOpComplete = function(type, context, data) {
+
+    /*
+     * Here we normalize all items to have a kind property.
```

Comment:
We've tried to not mutate data that comes from the server on the client. Is this used only for `makeId` in overview.js?